docs/user-guide.md

```diff
@@ -0,0 +1,856 @@
+  tags. Historically, this used to be the only way to reference your code.
+
+These should be inserted into the `<body>` of your HTML document.
+
+In both cases either use the `src` attribute to reference a Python
+file containing your code, or inline your code between the opening and closing
+tags. **We recommend you use the `src` attribute method**, but retain the
+ability to include code between tags for convenience.
+
+Here's the `<script>` tag method using the `src` attribute to reference the URL
+for a `main.py` Python file.
+
+```html title="Using the script tag with a source file"
+<script type="mpy" src="main.py"></script>
+```
+
+...and here's the `<py-script>` tag with inline Python code.
+
+```html title="Using the py-script tag with inline code"
+<py-script>
+import sys
+from pyscript import display
+
+
+display(sys.version)
+</py-script>
+```
+
+The browser's tab displaying the website running PyScript is an isolated
+computing sandbox. Define the Python environment in which your code will
+run with [configuration options](#configuration) (discussed later in this
+document).
+
+The `<script>` and `<py-script>` / `<mpy-script>` tags can have the following
+attributes:
+
+* `src` - the content of the tag is ignored and the Python code in the
+  referenced file is evaluated instead. **This is the recommended way to
+  reference your Python code.**
+* `config` - your code will only be evaluated after the referenced
+  configuration has been parsed. Since configuration can be JSON (or a TOML
+  file), `config='{"packages":["numpy"]}'` and `config="./config.json"` or
+  `config="./config.toml"` are all valid.
+* `async` - your Python code is run asynchronously (awaited), rather than in a
+  blocking manner.
+* `worker` - a flag to indicate your Python code is to be run on a web worker
+  instead of the "main thread" that looks after the user interface.
+
+!!! tip 
+
+    If you want to run code on both the main thread and in a worker, be
+    explicit and use separate tags.
+
+    ```html
+    <script type="mpy" src="main.py"></script>  <!-- on the main thread -->
+    <script type="py" src="worker.py" worker config="pyconfig.toml"></script> <!-- on the worker -->
+    ```
+
+    Notice how different interpreters can be used in different contexts.
+
+## Architecture
+
+There are two important aspects of PyScript's architecture:
+
+1. A small, efficient and powerful kernel called
+   [PolyScript](https://github.com/pyscript/polyscript) is the foundation
+   upon which PyScript and plugins are built.
+2. The PyScript [stack](https://en.wikipedia.org/wiki/Solution_stack) inside
+   the browser is relatively simple and easy to understand.
+
+### PolyScript
+
+[PolyScript](https://github.com/pyscript/polyscript) is the core of PyScript.
+
+!!! info
+
+    Unless you are an advanced user, you probably don't need to know much more
+    than PolyScript exists, and it can be safely ignored.
+
```

Comment:
are we sure we want to talk about PolyScript in this guide?
It's a very advanced topic, and I expect that 99% of our users will never need to know about it

Reply:
That's the point. I talk about it to tell folks, "you probably don't need to know about it" and then, as a courtesy, provide the bare minimum pointers for those few who do need to explore. :+1:

Reply:
Then I would put it at the end, in a section called "Advanced stuff that you don't need to know", or something like that.
+1 for adding a better explanation of the internal, but -0.5 to put it so early in the document, since most people are not interested in knowing about polyscript